src/io/flutter/inspector/WidgetPerfTipsPanel.java

```diff
@@ -22,23 +24,21 @@
 import io.flutter.perf.FlutterWidgetPerfManager;
 import io.flutter.perf.PerfTip;
 import io.flutter.perf.WidgetPerfLinter;
+import io.flutter.perf.*;
 import io.flutter.run.daemon.FlutterApp;
+import io.flutter.utils.AsyncUtils;
 import org.jetbrains.annotations.NotNull;
 
 import javax.swing.*;
+import javax.swing.Timer;
 import java.awt.event.ActionEvent;
-import java.util.ArrayList;
-
-// TODO(jacobr): display a table of all widget perf stats in this panel along
-// with the summary information about the current performance stats.
+import java.util.*;
 
 /**
- * Panel displaying basic information on Widget perf for the currently selected
- * file.
+ * Panel displaying performance tips for the currently visible files.
  */
-public class WidgetPerfPanel extends JPanel {
```

Comment:
Simplified this to just focus on the perf tips as the messages are now displayed inline in the editor.